### ISSUE TITLE
pm.sample cannot pickle LimbDark object

### ISSUE THREAD

**Issue by LionelGarcia** [exoplanet-dev/exoplanet#115]
**Error**
Getting `TypeError: cannot pickle 'exoplanet.theano_ops.driver.LimbDark' object` when sampling a model containing `LimbDarkLightCurve`

**Reproduction**
Fresh virtualenv with `exoplanet`, `pymc3`, `tqdm` installed 
```shell
python3 -m venv xotest
source ./xotest/bin/activate.fish
pip install tqdm exoplanet pymc3
```
and running [docs/tutorials/transit.py](https://github.com/exoplanet-dev/exoplanet/blob/main/docs/tutorials/transit.py)

```shell
cd xotest
curl -O https://raw.githubusercontent.com/exoplanet-dev/exoplanet/main/docs/tutorials/transit.py
python ./transit.py
```
**Setup:**
 - exoplanet==0.4.0
 - macOS 10.14.6
 - Python 3.8.5
 - pip 20.1.1

**Comment by dfm** [comment]
You won't be able to run that as a script on macOS with Python 3.8 because of issues with how multiprocessing now works. You'll need to wrap your script in a `if __name__ == "__main__"` and then make sure that multiprocessing is using `fork` instead of `spawn` or `forkserver`.

Unfortunately there's not much to do from within exoplanet except add a page to the docs so I'm going to relabel this as a documentation issue rather than a bug.

Related issues:

- https://github.com/pymc-devs/pymc3/issues/3844
- https://github.com/rodluger/starry/issues/254
- https://github.com/pymc-devs/pymc3/issues/4085
- https://github.com/exoplanet-dev/exoplanet/issues/111

**Comment by LionelGarcia** [comment]
I get this error as well on Linux when simply trying to pickle the model in [docs/tutorials/transit.py](https://github.com/exoplanet-dev/exoplanet/blob/main/docs/tutorials/transit.py). Like:

```python
with open("test.pickle", "wb") as f:
    pickle.dump(model, f)
```
`TypeError: cannot pickle 'exoplanet.theano_ops.driver.LimbDark' object`


It works with `exoplanet 0.3.2` but not `0.4.0`

**Comment by dfm** [comment]
OK - I'll take a look into this because it would be good to be able to pickle these models, but I don't understand why I haven't run into a sampling issue before if you're still getting it. I use parallel sampling on macos and ubuntu every day with v0.4 and the tutorial is automatically run on CI for every push.

**Comment by LionelGarcia** [comment]
This is puzzling indeed. I will try to investigate as well what could be wrong in my process.

_For reference:_ Before running into the last point I tried using `if __name__ == "__main__"` with `fork` (trying both `mp_ctx` and `multiprocessing.set_start_method` as mentioned in https://github.com/pymc-devs/pymc3/issues/3844#issuecomment-674091556) with Python 3.8.5 and Python 3.6.5 on macOS, leading to the same error.

**Comment by dfm** [comment]
This will all have something to do with running in a Jupyter environment vs a script. These tutorials are designed to be run using [jupytext](https://github.com/mwouts/jupytext), not directly as a script. #120 fixes the issue with pickling and #121 will track documenting best practices for running parallel sampling as a script (this is where the `__name__` part comes in!).
